src/main/java/dev/harrel/jsonschema/providers/SnakeYamlNode.java

```diff
@@ -0,0 +1,118 @@
+package dev.harrel.jsonschema.providers;
+
+import dev.harrel.jsonschema.JsonNode;
+import dev.harrel.jsonschema.JsonNodeFactory;
+import org.yaml.snakeyaml.LoaderOptions;
+import org.yaml.snakeyaml.Yaml;
+import org.yaml.snakeyaml.constructor.AbstractConstruct;
+import org.yaml.snakeyaml.constructor.SafeConstructor;
+import org.yaml.snakeyaml.nodes.*;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public final class SnakeYamlNode extends SimpleJsonNode {
+
+    private SnakeYamlNode(Object node, String jsonPointer) {
+        super(node, jsonPointer);
+    }
+
+    private SnakeYamlNode(Object node) {
+        this(node, "");
+    }
+
+    @Override
+    public boolean asBoolean() {
+        return (Boolean) node;
+    }
+
+    @Override
+    public String asString() {
+        return String.valueOf(node);
+    }
+
+    @Override
+    public List<JsonNode> asArray() {
+        List<?> list = (List<?>) node;
+        List<JsonNode> elements = new ArrayList<>(list.size());
+        for (Object o : list) {
+            elements.add(new SnakeYamlNode(o, jsonPointer + "/" + elements.size()));
+        }
+        return elements;
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public Map<String, JsonNode> asObject() {
+        Map<String, JsonNode> map = new HashMap<>();
+        for (Map.Entry<String, ?> entry : ((Map<String, ?>) node).entrySet()) {
+            map.put(entry.getKey(), new SnakeYamlNode(entry.getValue(), jsonPointer + "/" + JsonNode.encodeJsonPointer(entry.getKey())));
+        }
+        return map;
+    }
+
+    @Override
+    boolean isNull(Object node) {
+        return node == null;
+    }
+
+    @Override
+    boolean isArray(Object node) {
+        return node instanceof List;
+    }
+
+    @Override
+    boolean isObject(Object node) {
+        return node instanceof Map;
+    }
+
+    public static final class Factory implements JsonNodeFactory {
+        private final Yaml yaml;
+
+        public Factory() {
+            this.yaml = new Yaml(new JsonSchemaCompatibleConstructor());
+        }
```

Comment:
Yeah, initially it was that way but I felt it may do no good. If it's not properly configured it won't be complaint with JSON Schema spec. Still not sure though - do you feel it'd be useful? Do you heave some specific case already?

I liked the idea that users can reuse their own `Yaml` instances if they want to but it requires many customizations to make it right

Reply:
For v2 I would have a use case: to share my instance, just to be consistent with normal parsing versus validator-parsing.

> If it's not properly configured it won't be complaint with JSON Schema spec.

Is this because this factory also parses the Schema?

Reply:
> Is this because this factory also parses the Schema?

it applies to instance and schema documents:
- having non-string property names would break stuff
- floating point numbers are not parsed with enough precision (all parsed to Double)

so if any document has these it will break

Reply:
ofc allowing user to modify yaml factory and giving appropriate warning is also viable solution - but I'm still not sure if that's a good idea